docs/ref/checks.txt

```diff
@@ -239,6 +239,8 @@ Related Fields
 * **fields.E338**: The intermediary model ``<through model>`` has no field
   ``<field name>``.
 * **fields.E339**: ``<model>.<field name>`` is not a foreign key to ``<model>``.
+* **fields.E340**: Field is using a table that has already been registered
+  by ``<model>``.
```

Comment:
I'm not sure how to express it but the message should mention that it can be formulated as `<model>.<field>` when the conflicting model is `auto_created`.